pkg/client/client.go

```diff
@@ -233,18 +253,66 @@ func Initialize(
 		}
 	})
 
-	for _, application := range sanctionedApplications {
-		go checkStatusAndRegisterForApplication(ctx, ethereumChain, application)
+	blockCounter := hostChain.BlockCounter()
+
+	supportedApplicationsByStringID := make(map[string]chain.BondedECDSAKeepApplicationHandle)
+	tbtcApplicationHandle, err := hostChain.TBTCApplicationHandle()
+	if err != nil {
+		logger.Errorf(
+			"failed to look up on-chain tBTC application information for chain " +
+				"[%s]; tBTC application functionality will be disabled",
+		)
+	} else {
+		supportedApplicationsByStringID[tbtcApplicationHandle.ID().String()] =
+			tbtcApplicationHandle
+	}
```

Comment:
In case the `TBTCSystem` address is not set, `hostChain.TBTCApplicationHandle()` returns an error. If so, the `supportedApplicationsByStringID` map doesn't contain the handle and  `checkStatusAndRegisterForApplication` behavior is not triggered at all. Is this intentional behavior? My understanding is that if `TBTCSystem` is not set, only tBTC extensions don't work, not the entire client registration process.

Reply:
Hmmm, yes, that was my understanding as well, but you're right. We were previously leaning entirely on `SanctionedApplications` to manage the registration piece, and `Extensions.TBTC` to manage the extension. I think this approach is _strictly better_, and should deprecate `SanctionedApplications`---but it also means we need to deal with backwards-compatibility with `SanctionedApplications` being set without `Extensions.TBTC`.

Since currently-running operators should only have one entry in `SanctionedApplications` and it should match `TBTCSystem`, I'm going to suggest we adjust a couple of things:

- Check `SanctionedApplications` in addition to `Extensions.TBTC` for a fallback address.
- panic if both exist but don't match.
- Deprecate `SanctionedApplications`.

Thoughts?

Reply:
Yep sounds good!

Reply:
Ok, made these changes.